### PR TITLE
Using the Bokeh models API to create new charts

### PR DESCRIPTION
**Pull request by afausti**
In this technote we show how to use the Bokeh models API to create new charts, in particular we are interested in metric visualization for the LSST DM verification framework.

### CODE REVIEW

**Comment by jonathansick** [comment]
@afausti I think you need to rebase this branch against `master` in order to get the LTD configurations in your Travis file.

**Comment by afausti** [comment]
Thanks @jonathansick after the rebase I got the development edition for this branch.

File: index.ipynb [hunk 1]
```diff
@@ -0,0 +1,906 @@
...
+   "metadata": {},
+   "source": [
+    "### Setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This technical note is available as a Jupyter notebook from its [GitHub repository](https://github.com/lsst-sqre/sqr-022)."
```
Comment (SimonKrughoff) [inline index.ipynb]: Maybe link directly to the notebook (default to master perhaps)?

File: index.ipynb [hunk 2]
```diff
@@ -0,0 +1,906 @@
...
+    "                           'stretch': [5, 5, 10],\n",
+    "                           'design': [10, 10, 15],\n",
+    "                           'minimum': [20, 20, 30]})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we follow the *additive development* approach as suggested in the post. The idea is to explicitly add each Bokeh model as opposed to use the Bokeh plotting API and modify the plot defaults.\n",
```
Comment (SimonKrughoff) [inline index.ipynb]: use the --> using the
modify the --> modifying the

File: index.ipynb [hunk 3]
```diff
@@ -0,0 +1,906 @@
...
+   "outputs": [],
+   "source": [
+    "p1.x_range = Range1d(-100, 100)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In the y-axis we want to display the metric names, a categorial variable, so we use a `FactorRange()` for that, and initialize the factors with the categories present in our dataset. We also need to specify an appropriate scale for this axis."
```
Comment (SimonKrughoff) [inline index.ipynb:483]: Why do you need the `CategoricalScale`?  Isn't just having a `FactorRange` enough to make the plot work?

File: index.ipynb [hunk 4]
```diff
@@ -0,0 +1,906 @@
...
+    "p1.add_layout(percentage)\n",
+    "\n",
+    "show(p1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The colors indicate if the actual metric value satisfies the design specification (blue) or not (orange)."
```
Comment (SimonKrughoff) [inline index.ipynb:604]: Is this strictly true.  I think it's true if a smaller metric value is always better, but is it true for metrics where a larger value is better?

File: index.ipynb [hunk 5]
```diff
@@ -0,0 +1,906 @@
...
+    }
+   ],
+   "source": [
+    "p2.outline_line_color = 'white'\n",
+    "\n",
+    "height = 0.5\n",
+    "hbar = HBar(y='metrics', left='stretch', right='minimum', line_color='lightgray', \n",
+    "            fill_color='lightgray', height=height)\n",
+    "\n",
+    "p2.add_glyph(source, hbar)\n",
```
Comment (SimonKrughoff) [inline index.ipynb:698]: How does `add_glyph` know which y-values to use in plotting the HBar?

Comment (SimonKrughoff) [reply index.ipynb:698]: Oh never mind.  I see it's using the 'metrics' column for y.

**Review by SimonKrughoff — APPROVED** [review]
Excellent!  Thanks for putting this together.

File: index.ipynb (hunk 3 above)
Comment (afausti) [reply index.ipynb:483]: So, Bokeh requires in addition to the Range Model an appropriate Scale Model. The documentation says: "Scale models represent an invertible computation to be carried out on the client-side". There is this error if the Scale Model is not specified: ```ERROR:bokeh.core.validation.check:E-1009 (INCOMPATIBLE_SCALE_AND_RANGE): A Scale is incompatible with one or more ranges on the same plot dimension: incompatibility on y-dimension.```

File: index.ipynb (hunk 4 above)
Comment (afausti) [reply index.ipynb:604]: Good point, we can improve that using the operator defined as part of the specification.

**Comment by afausti** [comment]
Thanks for reviewing @SimonKrughoff

File: index.ipynb (hunk 3 above)
Comment (SimonKrughoff) [reply index.ipynb:483]: Can you add a very short mention of this.

**Pull request merged**